### ISSUE TITLE
Training data are wrongly oriented

### ISSUE THREAD

**Issue by jcohenadad** [ivadomed/model_seg_mice_gm-wm_dwi_unet2d-multichannel-softseg#10]
I checked these data: **duke:projects/ivadomed/gm_t2_exvivo** on 2021-03-31 at 21:22:32 and the data are A-P flipped, which could cause [this issue](https://github.com/neuropoly/spinalcordtoolbox/pull/3321#issuecomment-811567319) during inference:

![Screen Shot 2021-03-31 at 9 21 58 PM](https://user-images.githubusercontent.com/2482071/113230553-3c8b9700-9267-11eb-8990-05f824da4fbd.png)

A suggestion (to validate) would be to reorient the images and ground truth as follows:
~~~
sct_image -i IMAGE.nii.gz -setorient-data PSL -o IMAGE_NEW.nii.gz
~~~

**Comment by jcohenadad** [comment]
While reorienting the data i noticed inconsistent q/sform:

**sub-3902bottom_T2w_wmseg_manual.nii.gz**
~~~
qform_name	Aligned Anat
qform_code	2
qto_xyz:1	0.000000 0.000000 0.156250 -8.800000 
qto_xyz:2	-0.156250 0.000000 0.000000 10.300000 
qto_xyz:3	0.000000 -0.156250 0.000000 40.000000 
qto_xyz:4	0.000000 0.000000 0.000000 1.000000 
qform_xorient	Anterior-to-Posterior
qform_yorient	Superior-to-Inferior
qform_zorient	Left-to-Right
~~~

**sub-3902bottom_T2w_gmseg_manual.nii.gz**
~~~
qform_name	Unknown
qform_code	0
qto_xyz:1	0.000000 0.000000 0.156250 -8.800000 
qto_xyz:2	-0.156250 0.000000 0.000000 10.300000 
qto_xyz:3	0.000000 -0.156250 0.000000 40.000000 
qto_xyz:4	0.000000 0.000000 0.000000 1.000000 
qform_xorient	Anterior-to-Posterior
qform_yorient	Superior-to-Inferior
qform_zorient	Left-to-Right
~~~

A qform_code of 0 is undesired and could cause trouble when manipulating images. I will enforce the qform to match that of the image, which is different from the two labels:
~~~
qform_name	Scanner Anat
qform_code	1
qto_xyz:1	0.000000 0.000000 0.156250 -8.800000 
qto_xyz:2	-0.156250 0.000000 0.000000 10.300000 
qto_xyz:3	0.000000 -0.156250 0.000000 40.000000 
qto_xyz:4	0.000000 0.000000 0.000000 1.000000 
qform_xorient	Anterior-to-Posterior
qform_yorient	Superior-to-Inferior
qform_zorient	Left-to-Right
~~~

**Comment by jcohenadad** [comment]
Digging into the manual labels: there are additional consistencies:

the GM labels are rotated:
![image](https://user-images.githubusercontent.com/2482071/113232841-f258e480-926b-11eb-806b-86a61b5f6ade.png)

while the WM labels are correct:
![image](https://user-images.githubusercontent.com/2482071/113232871-fedd3d00-926b-11eb-8766-74a919925a6b.png)

**Comment by jcohenadad** [comment]
To fix the bottom GM labels, I had to do this:
~~~
sct_image -i sub-3902bottom_T2w_gmseg_manual.nii.gz -setorient-data LAI -o sub-3902bottom_T2w_gmseg_manual_LAI.nii.gz
~~~

After this step, both labels and the anat matched

**Comment by jcohenadad** [comment]
Now, looking at the top label, the problem is different: here both labels are matched, but they are AP flipped:

![image](https://user-images.githubusercontent.com/2482071/113233732-e1a96e00-926d-11eb-838e-f8a5895b6759.png)

To fix that I did:
~~~
# copy header from anat
sct_image -i ../../../../sub-3902top/anat/sub-3902top_T2w.nii.gz -copy-header sub-3902top_T2w_gmseg_manual.nii.gz -o sub-3902top_T2w_gmseg_manual_hd.nii.gz
# flip AP
sct_image -i sub-3902top_T2w_gmseg_manual_hd.nii.gz -setorient-data PSL -o sub-3902top_T2w_gmseg_manual_hd_PSL.nii.gz
~~~

I had to do it also for the top anat

ok, looking good now:
![image](https://user-images.githubusercontent.com/2482071/113234832-0acafe00-9270-11eb-817b-5c41f2aba86f.png)

and all qform/sform are the same:
~~~
qform_name	Scanner Anat
qform_code	1
qto_xyz:1	0.000000 0.000000 0.156250 -8.800000 
qto_xyz:2	-0.156250 0.000000 0.000000 10.300000 
qto_xyz:3	0.000000 -0.156250 0.000000 40.000000 
qto_xyz:4	0.000000 0.000000 0.000000 1.000000 
qform_xorient	Anterior-to-Posterior
qform_yorient	Superior-to-Inferior
qform_zorient	Left-to-Right
sform_name	Scanner Anat
sform_code	1
sto_xyz:1	0.000000 0.000000 0.156250 -8.800000 
sto_xyz:2	-0.156250 0.000000 0.000000 10.300000 
sto_xyz:3	0.000000 -0.156250 0.000000 40.000000 
sto_xyz:4	0.000000 0.000000 0.000000 1.000000 
sform_xorient	Anterior-to-Posterior
sform_yorient	Superior-to-Inferior
sform_zorient	Left-to-Right
~~~

**Comment by jcohenadad** [comment]
Another weird thing: under there are label files:
~~~
julien-macbook:~/temp/20210331/gm_t2_exvivo/sub-3902top/anat $ ll
total 27736
drwx------  6 julien  staff       192 31 Mar 21:54 .
drwx------  3 julien  staff        96 31 Mar 21:54 ..
-rwx------  1 julien  staff         2 31 Mar 21:54 sub-3902top_T2w.json
-rwx------  1 julien  staff  12150292 31 Mar 21:54 sub-3902top_T2w.nii.gz
-rwx------  1 julien  staff    696893 31 Mar 21:54 sub-3902top_T2w_gmseg.nii.gz
-rwx------  1 julien  staff   1342862 31 Mar 21:54 sub-3902top_T2w_wmseg.nii.gz
~~~

Hum... these are non-binary so I suspect they are generated by deepseg prediction. @charleygros can you confirm?

**Comment by charleygros** [comment]
> Hum... these are non-binary so I suspect they are generated by deepseg prediction. @charleygros can you confirm?

I confirm, they should not be here, can be deleted

**Comment by jcohenadad** [comment]
ok! data packaged here: **duke:projects/ivadomed/gm_t2_exvivo__20210331_223156.zip**

I suggest to upload that zip alongside the release of the next model.

**Comment by charleygros** [comment]
> 
> 
> ok! data packaged here: **duke:projects/ivadomed/gm_t2_exvivo__20210331_223156.zip**

Thanks so much! This info will be encoded in the config file. 

> I suggest to upload that zip alongside the release of the next model.

Sorry, what do you mean here? upload the dataset on this repo?

**Comment by charleygros** [comment]
> This info will be encoded in the config file.

[Here](https://github.com/ivadomed/model_seg_exvivo_gm-wm_t2_unet2d-multichannel-softseg/commit/0736072d712842c7b5d21773205369aa480c21a9).

**Comment by jcohenadad** [comment]
> Sorry, what do you mean here? upload the dataset on this repo?

As a release artifact:
![Screen Shot 2021-03-31 at 10 47 07 PM](https://user-images.githubusercontent.com/2482071/113236621-36031c80-9273-11eb-9755-e1a7855f62f5.png)

**Comment by charleygros** [comment]
Addressed with [this commit](https://github.com/ivadomed/model_seg_exvivo_gm-wm_t2_unet2d-multichannel-softseg/commit/5671890f10f1bea0d5d991ac766e0f5a1196f2d3).
And saved in the release r20210401_v2.
